sdk/data/azcosmos/cosmos_global_endpoint_manager.go

```diff
@@ -0,0 +1,170 @@
+		resourceAddress: "",
+	}
+
+	path, err := generatePathForNameBased(resourceTypeDatabaseAccount, "", false)
+	if err != nil {
+		return accountProperties{}, fmt.Errorf("failed to generate path for name-based request: %w", err)
+	}
+
+	azResponse, err := gem.client.sendGetRequest(path, context.Background(), operationContext, nil, nil)
+	if err != nil {
+		return accountProperties{}, fmt.Errorf("failed to send GET request: %w", err)
+	}
+
+	properties, err := parseAccountProperties(azResponse)
+	if err != nil {
+		return accountProperties{}, fmt.Errorf("failed to parse account properties: %w", err)
+	}
+
+	return properties, nil
+}
+
+// parseAccountProperties parses the account properties from the HTTP response.
+func parseAccountProperties(azResponse *http.Response) (accountProperties, error) {
+	properties := accountProperties{}
+	err := azruntime.UnmarshalAsJSON(azResponse, &properties)
+	if err != nil {
+		return properties, err
+	}
+	return properties, nil
+}
+
+// GetLocation returns the location for the given endpoint from the location cache.
+func (gem *globalEndpointManager) GetLocation(endpoint url.URL) string {
+	return gem.locationCache.getLocation(endpoint)
+}
+
+// MarkEndpointUnavailableForRead marks an endpoint as unavailable for read operations.
+func (gem *globalEndpointManager) MarkEndpointUnavailableForRead(endpoint url.URL) error {
+	return gem.locationCache.markEndpointUnavailableForRead(endpoint)
+}
+
+// MarkEndpointUnavailableForWrite marks an endpoint as unavailable for write operations.
+func (gem *globalEndpointManager) MarkEndpointUnavailableForWrite(endpoint url.URL) error {
+	return gem.locationCache.markEndpointUnavailableForWrite(endpoint)
+}
+
+// Update updates the location cache and the client's default endpoint based on the provided account properties.
+func (gem *globalEndpointManager) Update(accountProperties accountProperties) error {
+	if err := gem.locationCache.update(accountProperties.WriteRegions, accountProperties.ReadRegions, gem.preferredRegions, &accountProperties.EnableMultipleWriteLocations); err != nil {
+		return fmt.Errorf("failed to update location cache: %w", err)
+	}
+
+	gem.stopBackgroundRefresh()
+	gem.startBackgroundRefresh(accountProperties)
+
+	endpoints, err := gem.locationCache.writeEndpoints()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve write endpoints: %w", err)
+	}
+	if len(endpoints) > 0 {
+		gem.client.endpoint = endpoints[0].String()
+	}
+
+	return nil
+}
+
+func (gem *globalEndpointManager) startBackgroundRefresh(accountProperties accountProperties) {
+	gem.Lock()
+	defer gem.Unlock()
+
+	gem.refreshTicker = time.NewTicker(DefaultBackgroundRefreshInterval)
+	gem.refreshDone = make(chan struct{})
+
+	go func() {
+		for {
+			select {
+			case <-gem.refreshTicker.C:
+				err := gem.locationCache.update(accountProperties.WriteRegions, accountProperties.ReadRegions, gem.preferredRegions, &accountProperties.EnableMultipleWriteLocations)
```

Comment:
There is nothing calling GetAccountProperties to obtain a newer version of the account properties when the timer is up, shouldn't it be a call to `newAccountProperties:= gem.GetAccountProperties()` here? And then call `gem.Update(newAccountProperties)` which should internall handle the `gem.locationCache.update`?

Reply:
I see.

Reply:
Sorry, this comment was resolved but the code is still the original one. How is this triggering a new HTTP request and an update with the response?